draft-ietf-mls-protocol.md

```diff
@@ -419,8 +419,71 @@ Application Message:
 Terminology specific to tree computations is described in
 {{ratchet-tree-terminology}}.
 
-We use the TLS presentation language {{!RFC8446}} to
-describe the structure of protocol messages.
+## Presentation Langauge
+
+We use the TLS presentation language {{!RFC8446}} to describe the structure of
+protocol messages.  In addition to the base syntax, we add two additional
+features, the ability for fields to be optional and the ability for vectors to
+have variable-size length headers.
+
+### Optional Value
+
+An optional value is encoded with a presence-signaling octet, followed by the
+value itself if present.  When decoding, a presence octet with a value other
+than 0 or 1 MUST be rejected as malformed.
+
+~~~~~
+struct {
+    uint8 present;
+    select (present) {
+        case 0: struct{};
+        case 1: T value;
+    }
+} optional<T>;
+~~~~~
+
+### Variable-size Vector Headers
+
+In the TLS presentation language, vectors are encoded as a sequence of encoded
+elements prefixed with a length.  The length field has a fixed size set by
+specifying the minimum and maximum lengths of the encoded sequence of elements.
+
+In MLS, there are several vectors whose sizes vary over significant ranges.  So
+instead of using a fixed-size length field, we use a variable-size length using
+the variable-length integer encoding described in Section 16 of  {{?RFC9000}}.
+Instead of presenting min and max values, the vector description simply includes
+a `V`. For example:
+
+~~~~~
+struct {
+    uint32 fixed<0..255>;
+    opaque variable<V>;
+} StructWithVectors;
+~~~~~
```

Comment:
Was the intention to convert all vectors to variable length?
- If so, there are still some fixed-length left
- If not, how did you decide which to change?

Reply:
The example here is intended to have an old-style vector next to the new style, of course.

I did intend to change everything over, but you're correct that I missed a bunch with whatever regexp I used.  I'll take another pass.

Reply:
Looks like this was due to merge conflicts.  Fixed now in any case.